app/Http/Controllers/Three/SignupsController.php

```diff
@@ -19,18 +21,48 @@ class SignupsController extends ApiController
     protected $transformer;
 
     /**
+     * The signup service instance.
+     *
+     * @var Rogue\Services\SignupService
+     */
+    protected $signups;
+
+    /**
      * Create a controller instance.
      *
      * @return void
      */
-    public function __construct()
+    public function __construct(SignupService $signups)
     {
+        $this->signups = $signups;
+
         $this->transformer = new SignupTransformer;
 
         $this->middleware('auth.api');
     }
 
     /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(SignupRequest $request)
+    {
+        $transactionId = incrementTransactionId($request);
```

Comment:
Hmm, I'd really love to move this into a middleware. Let's consider pulling in the ticket for this next sprint.

Reply:
Great call! Here's the ticket for it: https://www.pivotaltracker.com/n/projects/2019429/stories/152511949